### PR TITLE
pass api_key as Authorization header

### PR DESCRIPTION
**Pull request by bishwahang**
* set api_key on Authorization headers by passing it as mandatory argument like
  braze_url
* re-record all the VCR request with new requests

As of May 2020, Braze has changed how we read API keys to be more
secure. Now API keys must be passed as a request header.
Braze will continue to support the api_key being passed through the
request body and URL parameters, but will eventually be sunset. Please
update your API calls accordingly.

### CODE REVIEW

**Comment by bishwahang** [comment]
@morgoth @jboltz88 
I am passing the `api_key` to the base class initializers `BrazeRuby::Rest::Base`(https://github.com/jboltz88/braze_ruby/pull/15/files#diff-86ef3181d045a2527662b11eb98267a2R10), and also initializing the connection with Authorization header already set on `BrazeRuby::Http`(https://github.com/jboltz88/braze_ruby/pull/15/files#diff-4341681b37f214570d457ea93537d194R31).

The message signature has only been changed on internal methods, and the public api still remains the same.
I have also re-recorded the whole VCR cassettes with new request and the specs passes.

File: spec/integrations/email_sync_spec.rb [hunk 1]
```diff
@@ -5,11 +5,11 @@
 RSpec.describe 'email sync' do
   describe 'unsubscribes' do
     it 'responds with unsubscribed emails', vcr: true do
-      response = api.email_unsubscribes(start_date: '2019-03-20', end_date: '2019-03-25')
+      response = api.email_unsubscribes(start_date: '2020-06-17', end_date: '2020-06-18')
       emails = JSON.parse(response.body)['emails']
 
       expect(emails.size).to eq(1)
-      expect(emails.first['email']).to eq('wojtek@test.com')
```
Comment (morgoth) [inline spec/integrations/email_sync_spec.rb:12]: ;-(

File: spec/support/vcr.rb [hunk 2]
```diff
@@ -9,7 +9,9 @@
   config.cassette_library_dir = 'spec/fixtures/responses'
 
   config.default_cassette_options = {
-    match_requests_on: [:method, :uri, :body]
+    match_requests_on: [:method, :uri, :body],
+    record: :once,
```
Comment (morgoth) [inline spec/support/vcr.rb:13]: :+1:

File: lib/braze_ruby/rest/canvas_details.rb [hunk 3]
```diff
@@ -6,14 +6,12 @@ class CanvasDetails < Base
       attr_reader :api_key, :params
```
Comment (morgoth) [inline lib/braze_ruby/rest/canvas_details.rb:6]: I guess you could also remove attr reader for `api_key` in all those places

Comment (bishwahang) [reply lib/braze_ruby/rest/canvas_details.rb]: Removed the redundant `attr_reader` and also removed  `api_key` being passed on some other http request call body/params.

**Comment by morgoth** [comment]
@bishwahang I released 0.4.0 with this change

**Pull request merged**